compiler/natives/src/net/http/http.go

```diff
@@ -99,6 +99,9 @@ func (t *XHRTransport) RoundTrip(req *Request) (*Response, error) {
 	}
 
 	select {
+	case <-req.Context().Done():
+		xhr.Call("abort")
+		return nil, <-errCh
```

Comment:
Not that I want you to change anything, just looking to learn. How did you decide between doing `return nil, <-errCh` and something like `return nil, ctx.Err()`? Did you consider the tradeoffs?

Reply:
Since abort may cause onabort to execute, the errCh channel needs to be drained.

Actually... The code also relies on abort not triggering onabort if the request already failed or succeeded.
But that assumption was already present, since it was already possible to call abort. I haven't checked. I think https://xhr.spec.whatwg.org/#the-abort()-method implies that this assumption is correct, but I have misunderstood WHATWG standards before, since it is an unfamiliar style of writing specs for me.